src/containers/SignUpForm.tsx

```diff
@@ -17,8 +17,8 @@ const SignUpForm = () => {
     <form
       onSubmit={(e) => {
         e.preventDefault();
-        if (username && email) actions.signUp(username, email);
-        history.push("/success");
+        if (username && email)
+          actions.signUp(username, email, () => history.push("/success"));
```

Comment:
now with the callback here, we can navigate to the success page only after we have indeed successfully created the account.